components/builder-db/src/functions/jobs.rs

```diff
@@ -25,3 +28,16 @@ sql_function! {
 sql_function! {
   fn job_graph_mark_failed(in_id: BigInt) -> Integer
 }
+
+sql_function! {
+  fn next_pending_job_v2(p_worker: Text, p_target: Text) -> jobs::JobRecord;
+}
+
+sql_function! {
+  fn insert_group_v3(root_project: Text, project_names: Array<Text>, project_idents: Array<Text>, p_target: Text)
+  -> jobs::GroupRecord;
+}
+
+sql_function! {
+  fn pending_groups_v1(count: Integer) -> Array<jobs::GroupRecord>;
+}
```

Comment:
Your tansaction commit gave me the idea that you could convert these sql functions to rust functions by mimicking the logic inside a transaction and then eliminate the `Record` types.

Reply:
Even though we may convert these functions to use models, there could be other problems in doing.
One particular I could think of doing updates row-by-row can be extremely slow.

We may differ this conversion (all the sql functions) until we find it does not impose any performance penalty, preferably in a separate PR.

Reply:
yeah that is certainly a good concern. I'm fine with keeping the functions if you think that is best.